boot/bootutil/src/loader.c

```diff
@@ -1583,7 +1583,7 @@ context_boot_go(struct boot_loader_state *state, struct boot_rsp *rsp)
 {
     size_t slot;
     struct boot_status bs;
-    int rc;
+    int rc = -1;
```

Comment:
I wonder if we should pick a specific err, since the error codes seem to be positive numbers. Even `BOOT_EBADIMAGE`. The only path I can see through the code where rc doesn't get set is if the build is configured for zero images.

Reply:
A similar patch was submitted before; I think sometimes `gcc` just confuses itself.

Reply:
I was just saying that in the even that gcc somehow got it right, and the code did fall through, we should probably return a meaningful error code. But, I guess I'm also ok with the code as is.